torch/fx/symbolic_trace.py

```diff
@@ -307,13 +307,140 @@ def forward(*args, **kwargs):
+# for wrapping during symbolic tracing
+_wrapped_fn_patch_table : Dict[_GlobalDictWrapper, Dict['str', Any]] = {}
+
+def _call_wrapped_func(orig_fn, args, kwargs):
+    """
+    Given an ``orig_function`` to invoke, search the args and kwargs for
+    a Proxy object. If there is one, emit a ``call_function`` node to
+    preserve the call to this leaf function directly. Otherwise, just
+    return the results of this function call, as this function is not
+    being traced.
+    """
+    proxy = None
+
+    def find_proxy(x):
+        nonlocal proxy
+        if isinstance(x, Proxy):
+            proxy = x
+
+    map_aggregate(args, find_proxy)
+    map_aggregate(kwargs, find_proxy)
+
+    if proxy is not None:
+        return proxy.tracer.create_proxy('call_function', orig_fn, args, kwargs)
+    else:
+        return orig_fn(*args, **kwargs)
+
+def _patch_wrapped_functions() -> Dict[_GlobalDictWrapper, Dict['str', Any]]:
+    """
+    Go through ``_wrapped_fn_patch_table`` and, for each frame object, wrap
+    the listed global functions in the `call_wrapped_func` wrapper. Returns
+    a map from frame to dict mapping name to the original global function.
+    """
+    orig_fns : Dict[_GlobalDictWrapper, Dict['str', Any]] = {}
+
+    for frame_dict, names in _wrapped_fn_patch_table.items():
+        orig_fns.setdefault(frame_dict, {})
+        orig_fns_this_frame = orig_fns[frame_dict]
+        for name in names:
+            orig_fn = frame_dict[name]
+            orig_fns_this_frame[name] = orig_fn
+
+            def scope(orig_fn):
+
+                def wrapped(*args, **kwargs):
+                    return _call_wrapped_func(orig_fn, args, kwargs)
+                return wrapped
+            frame_dict[name] = scope(orig_fn)
+
+    return orig_fns
+
+def _unpatch_wrapped_functions(orig_fns : Dict[_GlobalDictWrapper, Dict['str', Any]]):
+    """
+    Given the ``orig_fns`` dict that ``_patch_wrapped_functions``,
+    replace all of the global functions with the original global functions
+    that were there before symbolic tracing.
+    """
+    for frame_dict, to_revert in orig_fns.items():
+        for name, fn in to_revert.items():
+            frame_dict[name] = fn
+
+def wrap(fn_or_name : Union[str, Callable]):
+    """
+    This function can be called at global scope in a module to cause
+    references to the global function secified by `fn_name` to use
+    them in FX.
+
+        # foo/bar/baz.py
+        def my_custom_function(x, y):
+            return x * x + y * y
+
+        torch.fx.wrap('my_custom_function')
+
+        def fn_to_be_traced(x, y):
+            # When symbolic tracing, the below call to my_custom_function will be inserted into
+            # the graph rather than tracing it.
+            return my_custom_function(x, y)
+
+    Args:
+
+        fn_name (Union[str, Callable]): The function or name of the global function to insert into the
+            graph when it's called
+    """
+    if callable(fn_or_name):
```

Comment:
safer to check for `.__code__` directly. Not all callables are code (e.g. `__list__`)